luigi/tasks/genome_mapping/genome_mapping_tasks.py

```diff
@@ -0,0 +1,395 @@
+        return url
+
+    def run(self):
+        cmd = ('rm -f {path}/{chromosome}.gz && '
+               'wget {url} -O {path}/{chromosome}.gz && '
+               'gunzip -f {path}/{chromosome}.gz'
+               ).format(path=genome_mapping().genome(self.species),
+                        chromosome=self.chromosome,
+                        species=self.species,
+                        url=self.get_url())
+        status = subprocess.call(cmd, shell=True)
+        if status != 0:
+            raise ValueError('Failed to run gunzip: %s' % cmd)
+
+
+class DownloadGenome(luigi.WrapperTask):
+    """
+    A wrapper task to download all chromosomes in a genome.
+    """
+    species = luigi.Parameter()
+    division = luigi.Parameter()
+
+    def requires(self):
+        for chromosome in GetChromosomeList(species=self.species, division=self.division).output():
+            yield DownloadChromosome(species=self.species,
+                                     chromosome=chromosome,
+                                     division=self.division)
+
+
+class GenerateOOCfile(luigi.Task):
+    """
+    Generate an ooc file to speed up blat searches.
+    """
+    species = luigi.Parameter()
+    division = luigi.Parameter()
+
+    def requires(self):
+        return DownloadGenome(species=self.species, division=self.division)
+
+    def output(self):
+        genome_path = genome_mapping().genome(self.species)
+        return luigi.LocalTarget(os.path.join(genome_path, '11.ooc'))
+
+    def run(self):
+        cmd = ('cat {path}/*.fa > {path}/merged.fasta && '
+               'blat {path}/merged.fasta /dev/null /dev/null '
+                   '-makeOoc={ooc}-temp -stepSize=5 -repMatch=2253 -minScore=0 && '
+               'rm {path}/merged.fasta &&'
+               'mv {ooc}-temp {ooc}').format(
+                    path=genome_mapping().genome(self.species),
+                    ooc=self.output().path)
+        status = subprocess.call(cmd, shell=True)
+        if status != 0:
+            raise ValueError('Failed to run blat ooc: %s' % cmd)
+
+
+class BlatJob(luigi.Task):
+    """
+    Run blat to map a fasta file with RNAcentral sequences to a chromosome.
+    """
+    taxid = luigi.IntParameter()
+    species = luigi.Parameter()
+    fasta_input = luigi.Parameter()
+    chromosome = luigi.Parameter()
+    division = luigi.Parameter()
+
+    def requires(self):
+        return {
+            'fasta': CleanSplitFasta(taxid=self.taxid, species=self.species),
+            'ooc': GenerateOOCfile(species=self.species, division=self.division),
+        }
+
+    def run(self):
+        if 'empty-file' in self.fasta_input:
+            with self.output().open('w') as psl_output:
+                psl_output.write(' ')
+            return
+        genome_path, _ = os.path.split(self.chromosome)
+        cmd = ('blat -ooc={ooc} -noHead -q=rna -stepSize=5 '
+               '-repMatch=2253 -minScore=0 -minIdentity=95 '
```

Comment:
Why is `repMatch` so high? I looked at the blat documentation and 1024 would be the normal value for DNA.

Reply:
I use the same parameters as the UCSC online blat which worked better than 1024 in my testing:
https://genome-euro.ucsc.edu/FAQ/FAQblat#blat5
added this comment to the code